framework/dynamic.go

```diff
@@ -81,65 +81,136 @@ func (frame *Framework[VertexProp, EdgeProp, MsgType]) EnactStructureChanges(g *
 			// From the gathered set of consecutive adds, apply them.
 			if len(src.OutEdges) > didxStart {
 				val := frame.AggregateRetrieve(src)
-				frame.OnEdgeAdd(g, sidx, didxStart, val)
+				msgs := frame.OnEdgeAdd(g, sidx, didxStart, val)
+				if g.Undirected || g.SendRevMsgs {
+					for eidx := didxStart; eidx < len(src.OutEdges); eidx++ {
+						edge := src.OutEdges[eidx]
+						didx := edge.Destination
+						var msg MsgType
```

Comment:
I suggest using g.EmptyVal as the default value, as the nil value might be valid and meaningful in some cases. :)

Reply:
Good catch, yeah, that would be better.